internal/controller/serving/inferencegraph_controller_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serving
+
+import (
+	"context"
+
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	"github.com/kuadrant/authorino/api/v1beta2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/constants"
+	testutils "github.com/opendatahub-io/odh-model-controller/test/utils"
+)
+
+var _ = Describe("InferenceGraph Controller", func() {
+	Context("When reconciling a Serverless InferenceGraph", func() {
+		var testNs string
+		var exampleComUrl *apis.URL
+
+		exampleComUrl, _ = apis.ParseURL("https://example.com")
+		emptyUrl, _ := apis.ParseURL("")
+
+		buildInferenceGraph := func(name string) kservev1alpha1.InferenceGraph {
+			return kservev1alpha1.InferenceGraph{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:   testNs,
+					Name:        name,
+					Annotations: make(map[string]string),
+				},
+				Spec: kservev1alpha1.InferenceGraphSpec{
+					Nodes: map[string]kservev1alpha1.InferenceRouter{
+						"root": {
+							RouterType: kservev1alpha1.Sequence,
+							Steps: []kservev1alpha1.InferenceStep{
+								{StepName: "one", InferenceTarget: kservev1alpha1.InferenceTarget{
+									ServiceName: "isvc",
```

Comment:
maybe add multiple steps to make sure all URLs will be added in the auth config?

Reply:
The InferenceGraph always has a single URL regardless of how many ISVCs are referenced :-)

Reply:
cool